### PR TITLE
Always reject System.exit.

### PR DESCRIPTION
**Pull request by abayer**
There is never a valid reason for calling `System.exit(1)` in a
Jenkins Groovy script. Ever. So let's just permanently hack it into
always being rejected. Also, let's special case it in `ScriptApproval`
to not even show up as a pending signature.

### CODE REVIEW

**Comment by daniel-beck** [comment]
What about `Runtime#exit` and `Runtime#halt`?

**Comment by daniel-beck** [comment]
Honestly if something's blacklisted there's no reason to allow it in a sandbox. Want to shoot yourself in the foot, you need to be an admin for that.

**Comment by abayer** [comment]
I've added `Runtime#exit(int)` and `Runtime#halt(int)`, and generalized the approach a bit to serve as the basis for a longer-term perma-blacklist solution.

File: src/main/java/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/SandboxInterceptor.java [hunk 1]
```diff
@@ -127,10 +127,10 @@
 
             // Allow calling closure variables from a script binding as methods
             if (receiver instanceof Script) {
-                Script s = (Script)receiver;
+                Script s = (Script) receiver;
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/SandboxInterceptor.java]: Please avoid unrelated whitespace changes. Makes your diff hard to read.

File: src/main/java/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/GroovySandbox.java [hunk 2]
```diff
@@ -121,7 +122,10 @@ public Scope enter() {
             if (ExtensionList.lookup(RootAction.class).get(ScriptApproval.class) == null) {
                 return; // running in unit test, ignore
             }
-            ScriptApproval.get().accessRejected(x, _context);
+            String signature = x.getSignature();
+            if (signature != null && !StaticWhitelist.isPermanentlyBlacklisted(signature)) {
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/GroovySandbox.java]: Uh, what? No. You need to call `accessRejected` here even when `signature == null` (that would be an “unclassified” error).
